compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -157,45 +188,62 @@ abstract class PrepJSInterop extends plugins.PluginComponent
           } reporter.error(exp.pos, "You may not export a trait")
         }
 
-        enterScalaCls { super.transform(cldef) }
+        enterOwner(OwnerKind.NonEnumScalaClass) { super.transform(cldef) }
 
-      // Catch ValorDefDef in js.Any
-      case vddef: ValOrDefDef if inJSAny =>
+      // Module export sanity check (export generated in JSCode phase)
+      case modDef: ModuleDef =>
+        if (shouldPrepareExports)
+          registerModuleExports(modDef.symbol.moduleClass)
+
+        enterOwner(OwnerKind.NonEnumScalaMod) { super.transform(modDef) }
+
+      // Catch ValOrDefDef in js.Any
+      case vddef: ValOrDefDef if enclosingOwner is OwnerKind.RawJSType =>
         transformValOrDefDefInRawJSType(vddef)
 
+      // Exporter generation
+      case ddef: DefDef =>
+        if (shouldPrepareExports) {
+          // Generate exporters for this ddef if required
+          exporters.getOrElseUpdate(ddef.symbol.owner,
+              mutable.ListBuffer.empty) ++= genExportMember(ddef)
+        }
+        super.transform(tree)
+
       // Catch ValDefs in enumerations with simple calls to Value
-      case ValDef(mods, name, tpt, ScalaEnumValue.NoName(optPar)) if inScalaEnum =>
+      case ValDef(mods, name, tpt, ScalaEnumValue.NoName(optPar))
+          if enclosingOwner is OwnerKind.Enum =>
```

Comment:
OK I will revert to the old behavior with `anyEnclosingOwner is OwnerKind.Enum`.